docs/concepts.md

```diff
@@ -0,0 +1,320 @@
+
+### Example: Replicas
+
+Replicas are the unit that the KPA is scaling up and down. The responsiveness of the
+overall system depends in no small part on how quickly Replicas can become active and
+able to process incoming traffic. Replicas follow what is essentially a simple FSM
+(though not modeled as such):
+
+```
+                           +--------------------------------------+
+                           |                                      |
+                           |                                      V         
+ReplicaSource --> ReplicasLaunching --> ReplicasActive --> ReplicasTerminated
+```
+
+The diagram shows four possible transitions:
+
+* `begin_launch`, from ReplicaSource to ReplicasLaunching
+* `finish_launching`, from ReplicasLaunching to ReplicasActive
+* `terminate_launch`, from ReplicasLaunching to ReplicasTerminated
+* `terminate_active`, from ReplicasActive to ReplicasTerminated
+
+Time spent in the ReplicasLaunching stock is influential on system behaviour because
+Requests can continue to arrive while launching takes place. In the original designs
+this was substantially more detailed; for this generation of the design I chose to
+ignore the details while rebuilding the core simulator framework. A lot of improvements
+to simulation accuracy will probably come from breaking that Stock into finer detail. 
+
+Replicas are represented with the `ReplicaEntity`, a specialisation of Entity. The
+specialisation holds logic necessary to create and delete `Endpoints` in the Kubernetes
+API that the KPA consults to determine the current number of Replicas running.
+
+### Example: Requests
+
+Indirectly, Requests are signal that the Knative Pod Autoscaler is trying to respond
+to. In a traditional simulation these are called "arrivals". Unlike traditional
+simulation, the configuration of the simulated system varies throughout the simulation
+as the KPA changes its desired Replica count.
+
+The Movements graph for Requests is:
+ 
+```
+               +---------------------+
+               |                     |
+               V                     |
+TrafficSource -+-> RequestsBuffered -+-> RequestsProcessing --> RequestsComplete
+                                     |
+                                     +-> RequestsFailed
+```
+
+The diagram shows five possible Movements:
+
+* `arrive_at_buffer`, from TrafficSource to RequestsBuffered. These are scheduled
+  during the creation of the simulation
+* `send_to_replica`, from RequestsBuffered to RequestsProcessing
+* `complete_request`, from RequestsProcessing to RequestsComplete
+* `buffer_backoff`, from RequestsBuffered back into itself to simulate Activator behaviour
+* `exhausted_attempts`, from RequestsBuffered to RequestsFailed, representing a timeout
+
+In the happy path (`arrive_at_buffer`, `send_to_replica`, `complete_request`), this
+is a linear path. However, this is true only in aggregate: the exact path for each
```

Comment:
Maybe reword to "The happy path (`arrive_at_buffer`, `send_to_replica`, `complete_request`) is
linear."